weasyl/controllers/moderation.py

```diff
@@ -2,246 +2,205 @@
+        # Method
+        {"status": form.status, "violation": int(form.violation or -1), "submitter": form.submitter},
+        # Reports
+        report.select_list(request.userid, form),
+        macro.MACRO_REPORT_VIOLATION,
+    ]))
 
 
-class modcontrol_hide_(controller_base):
-    login_required = True
-    moderator_only = True
+@moderator_only
+@token_checked
+def modcontrol_closereport_(request):
+    form = request.web_input(reportid='', action='')
+    report.close(request.userid, form)
+    raise HTTPSeeOther(location="/modcontrol/report?reportid=%d" % (int(form.reportid),))
 
-    @define.token_checked
-    def POST(self):
-        form = web.input(name="", submission="", character="")
 
-        if form.submission:
-            moderation.hidesubmission(int(form.submission))
-        elif form.character:
-            moderation.hidecharacter(int(form.character))
+@moderator_only
+def modcontrol_contentbyuser_(request):
+    form = request.web_input(name='', features=[])
 
-        raise web.seeother("/modcontrol")
+    submissions = moderation.submissionsbyuser(request.userid, form) if 's' in form.features else []
+    characters = moderation.charactersbyuser(request.userid, form) if 'c' in form.features else []
+    journals = moderation.journalsbyuser(request.userid, form) if 'j' in form.features else []
 
+    return Response(define.webpage(request.userid, "modcontrol/contentbyuser.html", [
+        form.name,
+        sorted(submissions + characters + journals, key=lambda item: item['unixtime'], reverse=True),
+    ]))
 
-class modcontrol_unhide_(controller_base):
-    login_required = True
-    moderator_only = True
 
-    @define.token_checked
-    def POST(self):
-        form = web.input(name="", submission="", character="")
+@moderator_only
+@token_checked
+def modcontrol_massaction_(request):
+    form = request.web_input(action='', name='', submissions=[], characters=[], journals=[])
+    if form.action.startswith("zap-"):
+        # "Zapping" cover art or thumbnails is not a bulk edit.
+        if not form.submissions:
+            raise WeasylError("Unexpected")
+        submitid = int(form.submissions[0])
+        type = form.action.split("zap-")[1]
+        if type == "cover":
+            moderation.removecoverart(request.userid, submitid)
+        elif type == "thumb":
+            moderation.removethumbnail(request.userid, submitid)
+        elif type == "both":
+            moderation.removecoverart(request.userid, submitid)
+            moderation.removethumbnail(request.userid, submitid)
+        else:
+            raise WeasylError("Unexpected")
+        raise web.seeother("/submission/%i" % (submitid,))
 
-        if form.submission:
-            moderation.unhidesubmission(int(form.submission))
-        elif form.character:
-            moderation.unhidecharacter(int(form.character))
+    # TODO: Return a text response
+    return moderation.bulk_edit(
```

Comment:
I think this will fail because it's not a Response object and the view callable isn't configured with a renderer. It should probably be wrapped in `Response()` here.